crates/radix-engine-toolkit/src/transaction_types/traverser/auxiliary/trusted_worktop/handler_function_calls.rs

```diff
@@ -0,0 +1,230 @@
+                    self.add_new_instruction(false, None);
+                    self.worktop_content_tracker.enter_untracked_mode();
+                }
+                _ => self.unknown_function_call(),
+            }
+        } else if address.as_node_id().is_global_fungible_resource_manager() {
+            match function_name {
+                FUNGIBLE_RESOURCE_MANAGER_CREATE_WITH_INITIAL_SUPPLY_IDENT => {
+                    // resturns unknown resources put on worktop
+                    self.add_new_instruction(false, None);
+                    self.worktop_content_tracker.enter_untracked_mode();
+                }
+                _ => self.unknown_function_call(),
+            }
+        } else if address
+            .as_node_id()
+            .is_global_non_fungible_resource_manager()
+        {
+            match function_name {
+                NON_FUNGIBLE_RESOURCE_MANAGER_CREATE_WITH_INITIAL_SUPPLY_IDENT
+                | NON_FUNGIBLE_RESOURCE_MANAGER_CREATE_RUID_WITH_INITIAL_SUPPLY_IDENT =>
+                {
+                    // resturns unknown resources put on worktop
+                    self.add_new_instruction(false, None);
+                    self.worktop_content_tracker.enter_untracked_mode();
+                }
+                _ => self.unknown_function_call(),
+            }
+        } else if matches!(
+            address.as_node_id().entity_type(),
+            Some(EntityType::GlobalOneResourcePool)
+        ) {
+            match function_name {
+                ONE_RESOURCE_POOL_INSTANTIATE_IDENT => {
+                    self.add_new_instruction(true, None);
+                }
+                _ => self.unknown_function_call(),
+            }
+        } else if matches!(
+            address.as_node_id().entity_type(),
+            Some(EntityType::GlobalTwoResourcePool)
+        ) {
+            match function_name {
+                TWO_RESOURCE_POOL_INSTANTIATE_IDENT => {
+                    self.add_new_instruction(true, None);
+                }
+                _ => self.unknown_function_call(),
+            }
+        } else if matches!(
+            address.as_node_id().entity_type(),
+            Some(EntityType::GlobalMultiResourcePool)
+        ) {
+            match function_name {
+                MULTI_RESOURCE_POOL_INSTANTIATE_IDENT => {
+                    self.add_new_instruction(true, None);
+                }
+                _ => self.unknown_function_call(),
+            }
+        } else {
+            self.handle_global_generic_component_function_call(
+                address,
+                blueprint_name,
+                function_name,
+                args,
+            );
+        }
+    }
+
+    fn handle_global_generic_component_function_call(
+        &mut self,
+        address: &PackageAddress,
+        _blueprint_name: &str,
+        function_name: &str,
+        args: &ManifestValue,
+    ) {
+        if FAUCET_COMPONENT.as_node_id() == address.as_node_id() {
```

Comment:
This will always be false since the passed address is a `PackageAddress` while what you're comparing against is a `ComponentAddress`. You might want to use the `FAUCET_PACKAGE` here instead. Alternatively, you might not want to compare the `NodeId`s but the actual type.